src/Symfony/Component/Validator/Constraints/TypeValidator.php

```diff
@@ -33,23 +33,42 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        $type = strtolower($constraint->type);
-        $type = 'boolean' == $type ? 'bool' : $constraint->type;
-        $isFunction = 'is_'.$type;
-        $ctypeFunction = 'ctype_'.$type;
+        $types = [];
```

Comment:
I would keep the diff smaller like this:

```php
$types = (array) $constraint->type;

foreach ($types as $type) {
    $type = strtolower($type);
    $type = 'boolean' == $type ? 'bool' : $constraint->type;
    $isFunction = 'is_'.$type;
    $ctypeFunction = 'ctype_'.$type;
    if (\function_exists($isFunction) && $isFunction($value)) {
        return;
    } elseif (\function_exists($ctypeFunction) && $ctypeFunction($value)) {
        return;
    } elseif ($value instanceof $constraint->type) {
        return;
    }
}
```

Reply:
done